### PR TITLE
feat: Simplify Toolkit Initialization

### PR DESCRIPTION
**Pull request by kazo0**
closes https://github.com/unoplatform/uno.toolkit.ui/issues/269

## PR Type

What kind of change does this PR introduce?

- Feature

### CODE REVIEW

File: src/library/Uno.Toolkit.Material/MaterialToolkitTheme.cs [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+
+namespace Uno.Toolkit.UI.Material
+{
+	/// <summary>
+	/// Material (Material Design 3) styles for the controls in the Uno.Toolkit.UI library.
+	/// </summary>
+	public class MaterialToolkitTheme : ResourceDictionary
+	{
+		private bool _isColorOverrideMuted;
+		private bool _isFontOverrideMuted;
```
Comment (Xiaoy312) [inline src/library/Uno.Toolkit.Material/MaterialToolkitTheme.cs:24]:
```suggestion
		private bool _isOverridingColor;
		private bool _isOverridingFont;
```

Comment (Xiaoy312) [reply src/library/Uno.Toolkit.Material/MaterialToolkitTheme.cs:24]: pick one
```suggestion
		private bool _isUpdatingColorOverride;
		private bool _isUpdatingFontOverride;
```

File: src/library/Uno.Toolkit.Material/MaterialToolkitTheme.cs [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+			{
+				SetColorOverrideSilently(colorOverride);
+			}
+
+			if (fontOverride is { })
+			{
+				SetFontOverrideSilently(fontOverride);
+			}
+
+			UpdateSource();
```
Comment (Xiaoy312) [inline src/library/Uno.Toolkit.Material/MaterialToolkitTheme.cs:186]:
```suggestion
			SetColorOverrideSilently(colorOverride);
			SetFontOverrideSilently(fontOverride);

			UpdateSource();
```
and change the setters signature to: 
```diff
-private void SetFontOverrideSilently(ResourceDictionary fontOverride)
+private void SetFontOverrideSilently(ResourceDictionary? fontOverride)
```

File: samples/Uno.Toolkit.Samples/Uno.Toolkit.Samples.Shared/App.xaml.cs [hunk 3]
```diff
@@ -122,6 +125,9 @@ protected override async void OnLaunched(XamlLaunchActivatedEventArgs e)
 
 				splash.Content = _shell = BuildShell();
 				loadable.IsExecuting = false;
+
+
+				_window.Content = _shell = BuildShell();
```
[the remark this answers is not in the document]
Comment (kazo0) [reply samples/Uno.Toolkit.Samples/Uno.Toolkit.Samples.Shared/App.xaml.cs:130]: ah, forgot to undo my changes here

File: doc/material-getting-started.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+There is now a new, simplified method for initializing the Material Toolkit resources. It is recommended to use the new `MaterialToolkitTheme` going forward since `MaterialToolkitResources` has been deprecated.
+
+### Inheritance
+ResourceDictionary &#8594; MaterialToolkitTheme
+
+### Constructors
+| Constructor    | Description                                           |
+|----------------|-------------------------------------------------------|
+| MaterialToolkitTheme()         | Initializes a new instance of the `MaterialToolkitTheme` `ResourceDictionary`.       |
+| MaterialToolkitTheme(`ResourceDictionary?` colorOverride, `ResourceDictionary?` fontOverride)         | Initializes a new instance of the `MaterialToolkitTheme` `ResourceDictionary` and applies the given overrides       |
```
Comment (Xiaoy312) [inline doc/material-getting-started.md:19]: dont need to syntax highlight everything; too much is no different than no highlight
```suggestion
| MaterialToolkitTheme() | Initializes a new instance of the `MaterialToolkitTheme` resource dictionary. |
| MaterialToolkitTheme(ResourceDictionary? colorOverride, ResourceDictionary? fontOverride) | Initializes a new instance of the `MaterialToolkitTheme` resource dictionary` and applies the given overrides. |
```

File: doc/material-getting-started.md [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+1. Open an existing Uno project, or create a new Uno project using the `Multi-Platform App (Uno Platform)` template.
+2. In the Solution Explorer panel, right-click on your app's **App Code Library** project (`PROJECT_NAME.csproj`) and select `Manage NuGet Packages...`
+3. Install the [**`Uno.Toolkit.WinUI.Material`**](https://www.nuget.org/packages/Uno.Toolkit.WinUI.Material) package
+4. Add the resources to `AppResources.xaml`:
+	```xml
+	<ResourceDictionary>
+		<ResourceDictionary.MergedDictionaries>
+
+			<!-- Load Material Toolkit resources (also load base Uno.Material resources) -->
+			<MaterialToolkitTheme xmlns="using:Uno.Toolkit.UI" />
```
Comment (Xiaoy312) [inline doc/material-getting-started.md:43]: we should clarify whether if MaterialToolkitTheme contains MaterialTheme
like, do the user needs to init both or just the toolkit's one?

File: src/library/Uno.Toolkit.Material/MaterialToolkitTheme.cs [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+#if IS_WINUI
+		private const string ToolkitPackageName = "Uno.Toolkit.WinUI";
+		private const string ToolkitMaterialPackageName = "Uno.Toolkit.WinUI.Material";
+		private const string PackageNameSuffix = "WinUI";
+#else
+		private const string ToolkitPackageName = "Uno.Toolkit.UI";
+		private const string ToolkitMaterialPackageName = "Uno.Toolkit.UI.Material";
+		private const string PackageNameSuffix = "UWP";
+#endif
+		#region FontOverrideSource (DP)
```
Comment (Xiaoy312) [inline src/library/Uno.Toolkit.Material/MaterialToolkitTheme.cs:35]: just for consideration, lets use this for consistency?: https://github.com/Xiaoy312/bag-o-scripts/blob/main/DependencyProperty%20codegen.linq

Comment (Xiaoy312) [reply src/library/Uno.Toolkit.Material/MaterialToolkitTheme.cs:35]: and move dp changed handler outside

File: src/library/Uno.Toolkit.Material/MaterialToolkitTheme.cs [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+
+#if IS_WINUI
+		private const string ToolkitPackageName = "Uno.Toolkit.WinUI";
+		private const string ToolkitMaterialPackageName = "Uno.Toolkit.WinUI.Material";
+		private const string PackageNameSuffix = "WinUI";
+#else
+		private const string ToolkitPackageName = "Uno.Toolkit.UI";
+		private const string ToolkitMaterialPackageName = "Uno.Toolkit.UI.Material";
+		private const string PackageNameSuffix = "UWP";
+#endif
```
Comment (Xiaoy312) [inline src/library/Uno.Toolkit.Material/MaterialToolkitTheme.cs:34]:
```suggestion
#if IS_WINUI
		private const string ToolkitPackageName = "Uno.Toolkit.WinUI";
		private const string ToolkitMaterialPackageName = "Uno.Toolkit.WinUI.Material";
		private const string PackageNameSuffix = "WinUI";
#else
		private const string ToolkitPackageName = "Uno.Toolkit.UI";
		private const string ToolkitMaterialPackageName = "Uno.Toolkit.UI.Material";
		private const string PackageNameSuffix = "UWP";
#endif

		private bool _isOverridingColor;
		private bool _isOverridingFont;

```

File: doc/material-getting-started.md (hunk 5 above)
Comment (kazo0) [reply doc/material-getting-started.md:43]: oh thats supposed to say "also load**s the** base Uno.Material resources" :)

ill make it more clear

File: doc/cupertino-getting-started.md [hunk 8]
```diff
@@ -0,0 +1,81 @@
+# Uno Toolkit Cupertino Library
+
+The Uno.Toolkit.Cupertino library is available as NuGet packages that can be added to any new or existing Uno solution.
+
+## Installation
+
+> [!NOTE]
+> As of [Uno Platform 4.7](https://platform.uno/blog/uno-platform-4-7-new-project-template-performance-improvements-and-more/), the solution template of an Uno app has changed. There is no longer a Shared project (.shproj), it has been replaced with a regular cross-platform library containing all user code files, referred to as the **App Code Library** project. This also implies that package references can be included in a single location without the previous need to include those in all project heads.
```
Comment (agneszitte-nventive) [inline doc/cupertino-getting-started.md:8]:
```suggestion
> As of [Uno Platform 4.7](https://platform.uno/blog/uno-platform-4-7-new-project-template-performance-improvements-and-more/), the solution template of the Uno app has changed. There is no longer a Shared project (.shproj), it has been replaced with a regular cross-platform library containing all user code files, referred to as the **App Code Library** project. This also implies that package references can be included in a single location without the previous need to include those in all project heads.
```

File: doc/material-getting-started.md [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+| ColorOverrideSource             | string            | (Optional) Gets or sets a Uniform Resource Identifier that provides the source location of a ResourceDictionary containing overrides for the default Uno.Material Color resources                                            |
+| FontOverrideSource     | string      | (Optional) Gets or sets a Uniform Resource Identifier that provides the source location of a ResourceDictionary containing overrides for the default Uno.Material FontFamily resources            |
+
+## Installation
+
+> [!NOTE]
+> As of [Uno Platform 4.7](https://platform.uno/blog/uno-platform-4-7-new-project-template-performance-improvements-and-more/), the solution template of an Uno app has changed. There is no longer a Shared project (.shproj), it has been replaced with a regular cross-platform library containing all user code files, referred to as the **App Code Library** project. This also implies that package references can be included in a single location without the previous need to include those in all project heads.
```
Comment (agneszitte-nventive) [inline doc/material-getting-started.md:35]:
```suggestion
> As of [Uno Platform 4.7](https://platform.uno/blog/uno-platform-4-7-new-project-template-performance-improvements-and-more/), the solution template of the Uno app has changed. There is no longer a Shared project (.shproj), it has been replaced with a regular cross-platform library containing all user code files, referred to as the **App Code Library** project. This also implies that package references can be included in a single location without the previous need to include those in all project heads.
```

**Review by jeromelaban — APPROVED** [review]

**Pull request merged**